core/src/test/java/org/hibernate/ogm/backendtck/id/gen/CustomIdGeneratorTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.id.gen;
+
+import static org.hibernate.ogm.utils.OgmAssertions.assertThat;
+
+import java.util.List;
+
+import org.hibernate.id.IdentifierGenerator;
+import org.hibernate.ogm.utils.jpa.OgmJpaTestCase;
+
+import org.junit.Test;
+
+/**
+ * Test user defined {@link IdentifierGenerator}
+ */
+public class CustomIdGeneratorTest extends OgmJpaTestCase {
+
+	@Test
+	public void test() {
+		inTransaction( em -> {
+			em.persist( new CustomIdGeneratorEntity() );
+		} );
+
+		inTransaction( em -> {
+			List<CustomIdGeneratorEntity> resultList = em.createQuery( "from CustomIdGeneratorEntity" ).getResultList();
+			assertThat( resultList ).hasSize( 1 );
```

Comment:
I would prefer not having a check on the size because the library we use allows you to check the content of a collection using `assertThat( resultList ).containsExactly( ... )` or, if you are only interested to the  id (like in this case), `assertThat( resultList ).onProperty( "id" ).containsExactly( ...)`. This generates better messages if/when the test fails.

Reply:
By the way, if you expect only one value from the query, it makes sense to also use `.getSingleResult()` (it simpler to test the result this way)